firestore/bulkwriter.go

```diff
@@ -329,12 +337,13 @@ func (bw *BulkWriter) send(i interface{}) {
 				j.attempts++
 
 				// Do we need separate retry bundler?
-				if j.attempts < maxRetryAttempts {
+				_, isRetryable := batchWriteRetryCodes[codes.Code(s.Code)]
+				if j.attempts >= maxRetryAttempts || !isRetryable {
```

Comment:
nit: For readability, I think it would be clearer to revert the `if/else` blocks back to original and have the conditional statement be `if j.attempts < maxRetryAttempts && isRetryable` so that someone reading the code won't have to mentally apply Demorgan's Laws to understand what's going on.